File: java/BinarySearch.java
public class binarysearch {
    int Binary(int arr[],int search)
	{
		int start=0,mid=0;
		int end=(arr.length)-1;
		while(start<=end)
		{
			mid=start+(end-start)/2;
		
		if(search<arr[mid])
		{
			end=mid-1;
		}
		else if(search>arr[mid])
		{
			start=mid+1;
		}
		else
		{
			return mid;
		}
		}
		return -1;
		}

	public static void main(String[] args) {
		int arr[]= {1,2,3,4,5};
				
		int a=0;
		BinarySearch ol=new BinarySearch();
		a=ol.Binary(arr,3);
		System.out.println(a);
		}
	}
    

File: python/binary_search.py
def binarySearch(arr, x):
	low = 0
	high = len(arr) - 1
	mid = 0
	while low <= high:
		mid = (high + low) // 2
		if arr[mid] < x:
			low = mid + 1
		elif arr[mid] > x:
			high = mid - 1
		else:
			return mid
	return -1


# input array be like:2 5 6 4
arr = map(int,input("Enter the array:").split(" ") )
arr=list(arr)
x = int(input("Enter the number to find position:"))
result = binarySearch(arr, x)
if result != -1:
	print("Element is present at index", str(result))
else:
	print("Element is not present in array")
